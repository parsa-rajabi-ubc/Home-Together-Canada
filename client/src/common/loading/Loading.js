/**
 *
 * @Author:     Parsa Rajabi
 * @Created:    2021.2.20
 *
 * @Description: Loading Component
 *
 */

import SyncLoader from "react-spinners/SyncLoader";
import React from "react";
import PropTypes from "prop-types";

const LOADING_MSG = "Loading...";
const LOADING_COLOR = "#EC8937";

function Loading(props) {
    const {isLoading} = props;

    return (
        <section className={"text-center min-h-screen transform translate-y-1/2"}>
            <h1 className={"page-title font-normal"}>{LOADING_MSG}</h1>
            <SyncLoader loading={isLoading} color={LOADING_COLOR}/>
        </section>
    );
}

Loading.propTypes = {
    isLoading: PropTypes.bool.isRequired
}


export default Loading;