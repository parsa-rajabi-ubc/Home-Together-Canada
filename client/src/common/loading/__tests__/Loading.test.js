/**
 * @Author:     Parsa Rajabi
 * @Created:    2021.2.20
 *
 * @Description: Loading Test
 *
 */

import React from 'react';
import renderer from  'react-test-renderer'
import Loading from "../Loading";

describe('Loading', () => {
    describe('Container test', () => {
        it('should match snapshot test if the page is loading', () => {
            //given
            const props = {
                isLoading: true
            }
            //when
            const component = renderer.create(<Loading {...props}/>);
            const tree = component.toJSON();
            //then
            expect(tree).toMatchSnapshot();
        });
        it('should match snapshot test if the page is NOT loading', () => {
            //given
            const props = {
                isLoading: false
            }
            //when
            const component = renderer.create(<Loading {...props}/>);
            const tree = component.toJSON();
            //then
            expect(tree).toMatchSnapshot();
        });
    });
});