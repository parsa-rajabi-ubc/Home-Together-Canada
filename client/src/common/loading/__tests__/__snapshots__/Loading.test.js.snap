// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Container test should match snapshot test if the page is NOT loading 1`] = `
<section
  className="text-center min-h-screen transform translate-y-1/2"
>
  <h1
    className="text-gray-800 h1"
  >
    Loading...
  </h1>
</section>
`;

exports[`Loading Container test should match snapshot test if the page is loading 1`] = `
<section
  className="text-center min-h-screen transform translate-y-1/2"
>
  <h1
    className="text-gray-800 h1"
  >
    Loading...
  </h1>
  <span
    className="css-0"
  >
    <span
      className="css-ln2j55"
    />
    <span
      className="css-v3r7xv"
    />
    <span
      className="css-14rncnu"
    />
  </span>
</section>
`;
